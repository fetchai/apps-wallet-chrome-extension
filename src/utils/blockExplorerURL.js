import { MAINNET_BLOCKEXPLORER, NETWORKS_ENUM, STORAGE_ENUM, TESTNET_BLOCKEXPLORER } from '../constants'


/**
 * return block explorer url (with accounts path) based on whichever network is users selected network.
 *
 * @returns {string}
 */


const blockExplorerURL = (url_bit = 'accounts/') => {
  const network = localStorage.getItem(STORAGE_ENUM.SELECTED_NETWORK)
  if (network === NETWORKS_ENUM.TESTNET) return TESTNET_BLOCKEXPLORER + url_bit
  else if (network === NETWORKS_ENUM.MAINNET) return MAINNET_BLOCKEXPLORER + url_bit
  else if (network === NETWORKS_ENUM.LOCALHOST) return ''
  else throw new Error(`the network from storage : ${network} is not an expected network`)
}

export { blockExplorerURL }