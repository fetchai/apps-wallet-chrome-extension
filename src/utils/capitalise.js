const capitalise = s => {
  if (typeof s !== 'string' || s.length === 0) return ''
  return s.charAt(0).toUpperCase() + s.slice(1)
}

export { capitalise }
