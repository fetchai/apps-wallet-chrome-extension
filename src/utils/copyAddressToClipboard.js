
async function copyToClipboard (str) {
  return new Promise((resolve, reject) =>
  {
    navigator.clipboard.writeText(str).then(() => {
      resolve(true)
    }, () => {
      reject(false)
    })
  })
  }


export {copyToClipboard}