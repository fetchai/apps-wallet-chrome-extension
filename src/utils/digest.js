import { createHash } from 'crypto'

const digest = (address_raw) => {
  const hash_func = createHash('sha256')
  hash_func.update(address_raw)
  return hash_func.digest()
}

export { digest }
