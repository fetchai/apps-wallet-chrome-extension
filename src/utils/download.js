import Storage from '../services/storage'
import { KEY_FILE_NAME, STORAGE_ENUM } from '../constants'

/**
   * Causes download of encrypted key file as json file, taking key file from storage.
   *
   * @returns {Promise<void>}
   */

  const download = async () => {
    const json_str = await Storage.getItem(STORAGE_ENUM.KEY_FILE)
    const element = document.createElement('a')
    const file = new Blob([json_str], { type: 'text/plain' })
    element.href = URL.createObjectURL(file)
    element.download = KEY_FILE_NAME
    document.body.appendChild(element)
    element.click()
  }

  export { download }