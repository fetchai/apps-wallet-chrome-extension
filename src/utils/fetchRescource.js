/*global chrome*/
import { EXTENSION } from '../constants'

/**
 * Extensions cannot make CORs requests in content scripts therefore such requests are made in background script and results passed to content script.
 * This has same API as HTML5 Fetch Api https://www.chromium.org/Home/chromium-security/extension-content-script-fetches so we can easily switch to run in browser.
 *
 * @param input
 * @returns {Promise<unknown>}
 */

function proxyToBackground (input, init) {
  return new Promise((resolve, reject) => {
    chrome.runtime.sendMessage({ input, init }, messageResponse => {
      const [response, error] = messageResponse
      if (response === null) {
        reject(null)
      } else {
        // Use undefined on a 204 - No Content
        const body = response.body ? new Blob([response.body]) : undefined
        resolve(new Response(body, {
          status: response.status,
          statusText: response.statusText,
        }))
      }
    })
  })
}

/**
 * This is the switch that allows us to run in browser making requests, or background script we pass them to background.
 *
 * @param input
 * @param init
 * @returns {*}
 */
function fetchResource (input, init) {
  return EXTENSION ? proxyToBackground(input, init) : fetch(input, init)
}

export { fetchResource }
