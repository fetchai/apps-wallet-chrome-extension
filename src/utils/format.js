   const format = (val) => val.substring(0, 12) + "..."

export {format}