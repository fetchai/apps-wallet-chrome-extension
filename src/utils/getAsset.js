/*global chrome*/
import {EXTENSION} from "../constants";

/**
 * Different is returned URI for asset stored in assets folder
 * depending on if we run as an extension or in browser. If this throws an error on compilation
 * check whether your EXTENSION flag is correct for the environment you are running in.
 *
 * @param name
 * @returns {string}
 */
const getAssetURI = name => EXTENSION ? chrome.runtime.getURL("/assets/" + name) :  `./assets/${name}`

export {getAssetURI}