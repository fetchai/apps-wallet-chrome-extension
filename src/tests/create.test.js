import React from 'react'
import ReactDOM from 'react-dom'
import chrome from 'sinon-chrome'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Router, { clear } from '../services/router'
import Create from '../views/create'
import {
  click_button,
  form_write_value,
  STRONG_PASSWORD,
  STRONG_PASSWORD_2,
  untilPromises,
  WEAK_PASSWORD
} from './utils/helper'

import { mockAccountPageConstructor } from './utils/mockAccountPageConstructor'

const WEAK_PASSWORD_ERROR_MESSAGE = 'Weak password: password requires 14 characters including a number and an uppercase, lowercase and special character'
const PASSWORDS_DONT_MATCH_ERROR_MESSAGE = 'Passwords Don\'t Match'
const PASSWORD_REQUIRED_ERROR_MESSAGE = 'Password required'

describe.skip(':Initial', () => {

  beforeAll(() => {
    global.chrome = chrome
  })

  beforeEach(() => {
     jest.resetAllMocks();
     jest.clearAllMocks()
  })

  afterEach(() => {
    cleanup()
    clear()
  })

  test.skip('Create renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Create/>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test.skip('initial snapshot test', () => {
    const component = renderer.create(<Create/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('test empty password field outputs password required error message', () => {
    const { getByTestId } = render(<Create/>)
    form_write_value(getByTestId, 'create_password', '')
    click_button(getByTestId, 'create_submit')
    expect(getByTestId('create_output')).toHaveTextContent(PASSWORD_REQUIRED_ERROR_MESSAGE)
  })

  test('test weak password shows weak password error message', () => {
    const { getByTestId } = render(<Create/>)
    form_write_value(getByTestId, 'create_password', WEAK_PASSWORD)
    click_button(getByTestId, 'create_submit')
    expect(getByTestId('create_output')).toHaveTextContent(WEAK_PASSWORD_ERROR_MESSAGE)
  })

  test('test not matching passwords outputs not matching passwords error message', async () => {
    const { getByTestId } = render(<Create/>)
    form_write_value(getByTestId, 'create_password', STRONG_PASSWORD)
    form_write_value(getByTestId, 'create_password_confirm', STRONG_PASSWORD_2)

    await click_button(getByTestId, 'create_submit')
    expect(getByTestId('create_output')).toHaveTextContent(PASSWORDS_DONT_MATCH_ERROR_MESSAGE)
  })

  test('test multiple incorrect form submissions overwrites form error message', () => {
    const { getByTestId } = render(<Create/>)
    // initially we give it a not matching one
    form_write_value(getByTestId, 'create_password', STRONG_PASSWORD)
    form_write_value(getByTestId, 'create_password_confirm', STRONG_PASSWORD_2)
    click_button(getByTestId, 'create_submit')
    expect(getByTestId('create_output')).toHaveTextContent(PASSWORDS_DONT_MATCH_ERROR_MESSAGE)
    // then secondly we give no password
    form_write_value(getByTestId, 'create_password', '')
    click_button(getByTestId, 'create_submit')
    expect(getByTestId('create_output')).toHaveTextContent(PASSWORD_REQUIRED_ERROR_MESSAGE)
  })

  test('correctly filled in and submitted form creates account and redirects to accounts page', async () => {

    mockAccountPageConstructor()
    const { getByTestId, getAllByTestId } = render(<Router><Create/></Router>)
    form_write_value(getByTestId, 'create_password', STRONG_PASSWORD)
    form_write_value(getByTestId, 'create_password_confirm', STRONG_PASSWORD)

    const submit_button = getByTestId('create_submit')

    await fireEvent(
      submit_button,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }))

       await untilPromises()
    // tests the account component is now mounted by router.
    const app = getAllByTestId('account')
    expect(app.length).toBe(1)
  })

})