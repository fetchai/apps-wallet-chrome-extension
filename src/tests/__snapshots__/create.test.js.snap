// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:Initial initial snapshot test 1`] = `
<div
  className="OverlayMain"
  data-testid="create"
  id="my-extension-root-inner"
>
  <div
    className="OverlayMainInner"
  >
    <h1>
      Create account
    </h1>
    <hr />
    <form
      id="form"
    >
      <input
        className="large-button "
        data-testid="create_password"
        id="user_password"
        name="user_password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <input
        className=" large-button create-confirm-password "
        data-testid="create_password_confirm"
        id="user_password_confirm"
        name="user_password_confirm"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="password"
        value=""
      />
      <output
        className="create-output "
        data-testid="create_output"
        id="output"
        type="text"
      >
        
      </output>
      <div
        className="small-button-container"
      >
        <button
          className="small-button create-button"
          data-testid="create_back_button"
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <button
          className="small-button create-button"
          data-testid="create_submit"
          onClick={[Function]}
          type="submit"
        >
          Next
        </button>
      </div>
    </form>
  </div>
</div>
`;
